### PR TITLE
raftstore: move duplicated code into exists macro

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

## What have you changed? (mandatory)

raftstore: move duplicated code into exists macro

## What are the type of the changes? (mandatory)

- Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)

Exists unit tests

### CODE REVIEW

File: src/raftstore/store/fsm/batch.rs [hunk 1]
```diff
@@ -68,52 +68,33 @@ macro_rules! impl_sched {
...
+            fn shutdown(&self) {
+                // TODO: close it explicitly once it's supported.
+                // Magic number, actually any number greater than poll pool size works.
+                for _ in 0..100 {
+                    let _ = self.sender.send(FsmTypes::Empty);
+                }
+            }
         }
-    }
+    };
```
Comment (kennytm) [inline src/raftstore/store/fsm/batch.rs]: Why the semicolon? 😅

File: src/raftstore/store/fsm/batch.rs [hunk 2]
```diff
@@ -68,52 +68,33 @@ macro_rules! impl_sched {
...
         }
-    };
-}
-
-impl_sched!(NormalScheduler);
-impl_sched!(ControlScheduler);
 
-impl<N: Fsm, C> FsmScheduler for NormalScheduler<N, C> {
-    type Fsm = N;
+        impl<N: Fsm, C: Fsm> FsmScheduler for $name<N, C> {
```
Comment (kennytm) [inline src/raftstore/store/fsm/batch.rs]: You could use a where clause to keep the previous more relaxed constraint.  

```rust 
impl<N, C> FsmScheduler for $name<N, C> 
where 
    $fsm: Fsm,
{
```

File: src/raftstore/store/fsm/batch.rs (hunk 1 above)
Comment (lonng) [reply src/raftstore/store/fsm/batch.rs]: This semicolon comes from L71 in the previous code.

**Comment by lonng** [comment]
@BusyJay PTAL

**Pull request merged**